test/_init_tests.js

```diff
@@ -265,13 +265,33 @@ sinon.sandbox.create = function(config) {
   return sandbox;
 };
 
+function mockConsoleError() {
+  consoleSandbox = sinon.sandbox.create();
+  this.consoleMock = consoleSandbox.mock(console);
+  this.consoleMock.expects('error').never();
+}
+
+function checkForUnexpectedConsoleErrors() {
+  try {
+    this.consoleMock.verify();
+  } catch (e) {
+    const helpMessage =
+        'Your test resulted in an unexpected call to console.error.\n' +
+        '⤷ If the error is real, fix the code that generated it.\n' +
+        '⤷ If the error is expected:\n' +
+        '  ⤷ Call "this.consoleMock.expects(\'error\').withArgs' +
+            '(\'<message>\')" before the code containing the error.\n' +
+        '  ⤷ Call "this.consoleMock.expects(\'error\').never()" ' +
+            'after the code containing the error.\n';
```

Comment:
1) I'm not sure if you have add the expectation before the offending call to `console.error(…)`, since mocks are only verified in the `afterEach` hook. You can remove the "before the code containing the error" part, replace it with "inside your test". Test writers should be able to decide where they want to write the expectations :)

2) Does this `.never()` resetting work as expected? Again, since `.verify()` happens at the end, I think you're telling test writers to reset the expectation that they just wrote, before the mock gets verified. I think if you want the test to be very explicit it should be something like:
```javascript
…
this.consoleMock.expects('error').withArgs('foo', 'bar');
doStuff(); // calls console.error('foo', 'bar');
this.consoleMock.verify();
this.consoleMock.expects('error').never();
…
```
But this seems a bit _too_ explicit. Maybe instead we can suggest this:
```javascript
this.consoleMock.expects('error').once().withArgs('foo', 'bar');
```
(this way, the `.verify()` call at the end verifies the `.once()` constraint)

3) `<message>` is vague. I suggest adding a link to http://sinonjs.org/releases/v4.5.0/matchers/ (nit: can we dynamically replace the version number to always point to the Sinon version we're using?)

Reply:
Based on offline discussions and debugging with @danielrozenberg, I retract my comment above re: abandoning the approach that uses mocks. Updated commits coming up. The above comments are moot.